hazelcast/src/main/java/com/hazelcast/nio/tcp/SocketReader.java

```diff
@@ -16,12 +16,62 @@
 
 package com.hazelcast.nio.tcp;
 
-import java.nio.ByteBuffer;
+import com.hazelcast.util.counters.Counter;
 
 /**
- * Reads content from a {@link ByteBuffer}.
+ * The SocketReader is responsible for reading data from the socket, on behalf of a connection, into a byte-buffer. Once the
+ * data is read into the ByteBuffer, this ByteBuffer is passed to the {@link ReadHandler} that takes care of the actual
+ * processing of the incoming data.
+ *
```

Comment:
I added it to provide some background on how the the classes are connected. So perhaps it isn't visible in the methods of the interface, this is the way this interface is going to be used. It is inherently tied to these classes.